web/src/components/graph/timeaxis.js

```diff
@@ -1,35 +1,40 @@
 import React from 'react';
-import moment from 'moment';
-import { range } from 'lodash';
+import Moment from 'moment';
+import { extendMoment } from 'moment-range';
 
 import { __ } from '../../helpers/translation';
 
-// If the tick represents a timestamp not more than 15 minutes in the past,
-// render it as "Now", otherwise render as localized time, i.e. "8:30 PM".
-const renderTickValue = v => (
-  moment().diff(moment(v), 'minutes') <= 15
+const moment = extendMoment(Moment);
+
+const renderTickValue = (v, format) => (
+  moment().diff(v, 'hours') < 1
     ? __('country-panel.now')
-    : moment(v).format('LT')
+    : v.format(format)
 );
 
-const roundUp = (number, base) => Math.ceil(number / base) * base;
-
-// Return `count` timestamp values uniformly distributed within the scale
-// domain, including both ends, rounded up to 15 minutes precision.
-const getTicksValuesFromTimeScale = (scale, count) => {
-  const startTime = scale.domain()[0].valueOf();
-  const endTime = scale.domain()[1].valueOf();
-
-  const precision = moment.duration(15, 'minutes').valueOf();
-  const step = (endTime - startTime) / (count - 1);
-
-  return range(count).map(ind => (
-    moment(ind === count - 1 ? endTime : roundUp(startTime + ind * step, precision)).toDate()
-  ));
-};
-
 const TimeAxis = React.memo(({ className, scale, transform }) => {
   const [x1, x2] = scale.range();
+
+  const startMoment = moment(scale.domain()[0]);
+  const endMoment = moment(scale.domain()[1]);
+  let interval;
+  let format;
+  if (endMoment.diff(startMoment, 'years') > 1) {
+    interval = 'year';
+    format = 'YYYY';
+  } else if (endMoment.diff(startMoment, 'months') > 1) {
+    interval = 'month';
+    format = 'MMM YYYY';
+  } else {
+    interval = 'hour';
+    format = 'LT';
+  }
+  const range = moment.range(
+    startMoment.startOf(interval).add(1, interval), // add one at the end to make sure it includes NOW
+    endMoment,
+  );
+  const tickMoments = Array.from(range.by(interval, { step: interval === 'hour' ? 6 : 1 }));
```

Comment:
Given how prone to errors / imprecisions managing scale ticks generally tends to be, it'd be great if we could keep this logic for generating `tickMoments` somewhat extracted and _unit tested_. Probably out of scope of this PR, but maybe you could at least add a `TODO` comment here above :wink: